### PR TITLE
options to opt out dependencies install and customize qt configuration

### PR DESCRIPTION
**Pull request by snakelizzard**
Added two new options:
* `requirements` default true - needed to opt out dependencies install if you are going to make build without gui
* ` config` default "" - provides additional options passed to configure script, mostly needed to force -no-feature-xxx

This patch also makes cross-compile from linux to arm available.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
Thanks, It would be nice if you could split this PR in 3:
1. one for the "config" option which is definitely a must have
2. one for the "requirements" part, which I think could be improved: the recipe has to be intelligent enough to know if they are necessary or not. In which case do you have to disable the requirements ?
3. one for the cross compilation to ARM

**Comment by snakelizzard** [comment]
@ericLemanissier Will do this later, too busy with my current tasks. In case I will not return to this or a long time, I will add comments to my commit to clarify what's what. Disabling requirements needed for builds without gui for arm devices. More clever way to do this is just install this packages only for linux.x86 and linux.x86_64.

**Comment by lucienboillod** [comment]
Well no problem @snakelizzard  we will split it. Thank you a lot for this contribution it helped a lot! :blush:

File: conanfile.py [hunk 1]
```diff
@@ -210,9 +216,17 @@ def _build_mingw(self, args):
 
     def _build_unix(self, args):
         if self.settings.os == "Linux":
-            args += ["-silent", "-xcb"]
-            if self.settings.arch == "x86":
-                args += ["-platform linux-g++-32"]
+            args += ["-silent"]
```
Comment (snakelizzard) [inline conanfile.py:219]: Keep -xcb option only for Linux x86 and x86_64

File: conanfile.py [hunk 2]
```diff
@@ -45,15 +46,18 @@ def getsubmodules():
...
     no_copy_source = True
-    default_options = ("shared=True", "fPIC=True", "opengl=no", "openssl=no") + tuple(module + "=False" for module in submodules)
+    default_options = ("shared=False", "fPIC=True", "opengl=no", "openssl=no", "requirements=True", \
+        "config=\"\"") + tuple(module + "=False" for module in submodules)
     short_paths = True
     build_policy = "missing"
 
     def build_requirements(self):
-        if tools.os_info.is_linux:
+        if tools.os_info.is_linux and self.options.requirements:
```
Comment (snakelizzard) [inline conanfile.py:60]: Opt out requirements install needed for core-only build, without qt gui, widgets, etc.

**Comment by ericLemanissier** [comment]
Ok, so instead of the "requirement" option, I am currently testing the addition of two boolean options : "GUI" and "widgets" which are forwarded to configure. If none of these options is activated, we simply skip all the xcb parts in the following methods: build_requirements, requirements and _build_unix

**Comment by ericLemanissier** [comment]
the GUI, widgets and config options have been added in Qt/5.11.1@bincrafters/stable
Also, architecture armv6 and armv7 now use correct platform argument when calling configure
the merge in 5.6.3 and 5.9.6 branches is ready, awaiting approval

**Comment by lucienboillod** [comment]
The changes has been pushed, Thanks a lot for all your proposal and your contribution it helped a lot!

**Pull request closed without merging**